backend/authapp/utils.py

```diff
@@ -1,15 +1,53 @@
-def can_modify(request, instance):
+from utils.general import str2bool
+
+from .exceptions import ModifyUserError
+
+def check_administrator_modify_restrictions(request, instance):
+    """
+    Checks if administrators are able to modify certain privileges of other administrator accounts
+
+    Cases
+    1. username admin (different from normal administrator accounts) privileges cannot be taken away
+    2. administrators cannot take away their own privileges
+    """
     modifying_user = request.user.username
     modified_user = instance.username
-    modified_user_is_staff = instance.is_staff
 
-    if modified_user == 'admin':
-        return False, "Cannot change admin(SuperUser) account privileges"
+    modifying_user_is_staff = str2bool(request.data.get('is_staff', 'true'))
 
-    if modifying_user == modified_user:
-        return False, "Cannot change self account privileges"
+    # Case 1
+    if modified_user == 'admin' and modifying_user_is_staff == False:
```

Comment:
Still a bit confused here. Won't the modifying user always be staff because of how this method is being used?